3660 integration tests

meandavejustice opened the pull request:
fixes #3660 

Attempting to fix integrations tests which were broken after addon started being built for the `dev` environment on circleci.

Commit:
try changing hosts file for integration tests to dev so we don't
have to build 2 addons

- fixes #3660

Commit:
uncomment install tests

Commit:
make sure we set cookies to correct domain for tests

Commit:
remove port 8000 in tox.ini

Commit:
try building local addon for integration tests

Commit:
try storing addon artifacts along with ui test results to help with debugging

Commit:
fix cookie setting back to example.com

b4handjr commented on frontend/test/ui/test_install.py:
```diff
@@ -1,36 +1,36 @@
+@pytest.mark.nondestructive
+@pytest.mark.skipif(os.environ.get('SKIP_INSTALL_TEST') is not None,
+                    reason='Skip install on Release and Beta Firefox.')
+def test_install_of_test_pilot_addon(
```
Looking at the build log I have some thoughts:

It looks like there is some sort of popup that is shown on the html report within circleci/artifacts. The test is kind of just looking for the standard webextension install notification.

b4handjr commented on frontend/test/ui/test_install.py:
```diff
@@ -45,53 +45,53 @@ def test_bottom_install_button(base_url, selenium, firefox, notifications):
+@pytest.mark.skipif(os.environ.get('SKIP_INSTALL_TEST') is not None,
+                    reason='Skip install on Release and Beta Firefox.')
+@pytest.mark.nondestructive
+def test_install_and_enable(base_url, selenium, firefox, notifications):
```
Looking the log it seems as if the Dev Example gets enabled but ```experiment.install_and_enable()``` is looking for something else. Specifically its within the ```self.wait.until()``` found in that class. I'd check there and make sure whatever it is waiting for matches the new page/styling.

b4handjr replied:
Maybe this got changed? https://github.com/mozilla/testpilot/blob/master/frontend/test/ui/pages/desktop/detail.py#L45

meandavejustice replied:
Seems to be the same, was just moved into ExperimentButtons https://github.com/mozilla/testpilot/blob/master/frontend/src/app/containers/ExperimentPage/ExperimentButtons.js#L143

b4handjr commented on frontend/test/ui/test_install.py:
```diff
@@ -45,53 +45,53 @@ def test_bottom_install_button(base_url, selenium, firefox, notifications):
+@pytest.mark.nondestructive
+@pytest.mark.skipif(os.environ.get('SKIP_INSTALL_TEST') is not None,
+                    reason='Skip install on Release and Beta Firefox.')
+def test_enable_and_disable_experiment(
```
This looks like the same error as above, please see my comments there.

Commit:
change dev example to  be of addon type for integration test

Commit:
Remove test for installing from featured button and header

Install test for bottom tests the same functionality without
the complications.

- fixes #3660

Commit:
change email popup loccator
